operator-common/src/main/java/io/strimzi/operator/common/model/Ca.java

```diff
@@ -525,12 +524,15 @@ public void createRenewOrReplace(
         int caKeyGeneration = keyGeneration();
 
         if (!generateCa) {
-            certData = caCertSecret != null ? caCertSecret.getData() : emptyMap();
-            keyData = caKeySecret != null ? singletonMap(CA_KEY, caKeySecret.getData().get(CA_KEY)) : emptyMap();
+            if (caCertSecret == null || caKeySecret == null)   {
+                throw new InvalidResourceException(this + " should not be generated, but the secrets were not found.");
+            }
```

Comment:
An empty line after the block would be nice. But I think it is strange and error-prone to have a validation of something passed to a constructor and stored in the class of a class as part of a method that is called separately. I think it is fine not to have it outside, but it should be done in the constructor and not in some method called only later.

Reply:
I've moved it to the constructor